spec.bs

```diff
@@ -1159,12 +1373,208 @@ fields:
+    discusses storage type of BBKs including additional topics of which storage
+    types should be allowed and exposing the storage type to the relying party.
+
+## Binding a key pair ## {#sctn-binding-a-keypair}
+
+To <dfn>bind a key pair</dfn> given the |bbk_id|, the key pair identifier, and
+|credential_id|, the SPC credential_id (i.e. passkey id), store the
+identifiers in the [=browser_bound_map=] as follows:
+
+1. Insert the browser bound key id into the map:
+    [=browser_bound_map=][|credential_id|] = |bbk_id|.
+
+    Note: When user agents are storing this map (such as serializing to disk)
+        and the procedure fails, the user agent may return an error
+        code from here and avoid including the browser bound key in which case
+        the relying party does not get the browser bound key on this
+        transaction. Or the user agent may continue in which case the browser
+        bound key is only used once. In either case, on a subsequent payment
+        assertion a new key pair will be created.
+
+## Retrieving a key pair ## {#sctn-retrieving-a-browser-bound-key}
+
+To <dfn>get or create a browser bound key</dfn> given |credential_id|, a
+[=Credential ID=], and |allowed_algorithms|, a
+{{PublicKeyCredentialParameters}} object perform the following steps that find
+an existing associated key pair or create a new associated key pair, and
+return that key pair.
+
+1. If [=browser_bound_map=][|credential_id|]
+
+    <dl class="switch">
+
+        : Exists
+        :: Let |bbk| be the key pair retrieved using the bbk id from
+            [=browser_bound_map=][|credential_id|].
+
+        : Does not exist
+        :: Create and bind a new key pair
+
+            1. Let (|bbk|, |bbk_id|) be the result of the procdure to [=create
+                a key pair=] using |allowed_algorithms|.
+
+            1. Bind the key pair to the SPC credential by [=binding a key
+                pair=] using |bbk_id| and |credential_id|.
+
+    </dl>
+
+1. Return |bbk|.
+
+## Getting the browser bound public key ## {#sctn-getting-the-bbk-public-key}
+
+To <dfn>get a browser bound public key</dfn> given the |bbk|, the key pair
+returned by the precedures above:
+
+1. Retrieve the |public_key| of the |bbk| according to established procedures
+    for the relevant algorithm.
+
+1. Let |encoded_public_key| be the COSE_Key encoding of |public_key|. Refer to
+    [=credentialPublicKey=] from WebAuthn's
+    [[webauthn-3#sctn-attested-credential-data]].
+
+1. Return |encoded_public_key|.
+
+## Signing client data ## {#sctn-signing-client-data}
+
+To <dfn>generate a browser bound signature</dfn> given the |bbk| and
+|client_data_json|, a byte array containing the encoded client data, perform
+the following steps:
+
+1. Generate |signature|, a byte array, by performing the cryptographic
+    signature algorithm on |client_data_json|. This corresponding algorithm was
+    chosen when [=creating a key pair=].
```

Comment:
You have to clarify where the "cryptographic signature algorithm" comes from. Either pass it in, store it somewhere outside this algorithm that this algorithm has access to, etc. It's too "magical" as-is.